pkgs/development/tools/build-managers/cmake/setup-hook.sh

```diff
@@ -43,17 +51,21 @@ cmakeConfigurePhase() {
     # libraries are in a system path or in the same directory as the
     # executable. This flag makes the shared library accessible from its
     # nix/store directory.
-    cmakeFlags="-DCMAKE_INSTALL_NAME_DIR=$prefix/lib $cmakeFlags"
-    cmakeFlags="-DCMAKE_INSTALL_LIBDIR=${!outputLib}/lib $cmakeFlags"
-    cmakeFlags="-DCMAKE_INSTALL_INCLUDEDIR=${!outputDev}/include $cmakeFlags"
+    cmakeFlags+=(
+        "-DCMAKE_INSTALL_NAME_DIR=$prefix/lib"
+        "-DCMAKE_INSTALL_LIBDIR=${!outputLib}/lib"
+        "-DCMAKE_INSTALL_INCLUDEDIR=${!outputDev}/include"
+    )
 
     # Avoid cmake resetting the rpath of binaries, on make install
     # And build always Release, to ensure optimisation flags
-    cmakeFlags="-DCMAKE_BUILD_TYPE=Release -DCMAKE_SKIP_BUILD_RPATH=ON $cmakeFlags"
-
-    echo "cmake flags: $cmakeFlags ${cmakeFlagsArray[@]}"
+    cmakeFlags+=(
+        "-DCMAKE_BUILD_TYPE=Release"
```

Comment:
There is an ongoing patch that changes this part: https://github.com/NixOS/nixpkgs/pull/17828